forumone/wp-cfm
# fix(PHP): Fixes PHP related deprecation notices & build bugs

### All Submissions:

* [x] Have you followed the [plugin Contributing guideline](https://github.com/forumone/wp-cfm/blob/develop/CONTRIBUTING.md)?





### Changes proposed in this Pull Request:



* Fixes #135
  * Fixes `FILTER_SANITIZE_STRING` deprecation notices using polyfill function.
* Fixes build related ignores and linting setup issues.
* Fixes the ability to generate a release `dist` regardless of the branch you are on.
* Fixes GitHub Actions workflows to ensure correct build environment setup and WP.org deployments.
* Adds feature/fix branch and `develop` code checks.
* Fixes #147
  * Fixes trim deprecation notices using newer Symfony/YAML package & `null` check before parsing YAML files.

### How to test the changes in this Pull Request:

1. Checkout the PR.
2. Generate a build of the plugin. `npm ci && npm run build`
3. Generate a release of the plugin. `npm run release`
4. Copy the files in the `dist` directory to a WordPress installation.
5. Attempt to load WP-CFM bundles from the WP-CLI. `wp config pull all`
6. From the WordPress Dashboard ensure that creating bundles, saving bundles, pull/push bundles all function without errors or notices in the debug logs.

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
* Fixes `FILTER_SANITIZE_STRING` deprecation notices using polyfill function.
* Fixes trim deprecation notices using latest Symfony/YAML package & `null` check patch.
